tests/generic_relations/tests.py

```diff
@@ -501,14 +500,27 @@ def test_query_content_type(self):
         with self.assertRaisesMessage(FieldError, msg):
             TaggedItem.objects.get(content_object="")
 
-    def test_unsaved_instance_on_generic_foreign_key(self):
-        """
-        Assigning an unsaved object to GenericForeignKey should raise an
-        exception on model.save().
-        """
+    def test_unsaved_instance_on_generic_foreign_key_save(self):
+        # Assigning an unsaved object to a GFK should raise a ValueError on save.
+        quartz = Mineral(name="Quartz", hardness=7)
+        tagged_item = TaggedItem(tag="shiny", content_object=quartz)
+        msg = (
+            "save() prohibited to prevent data loss due to unsaved related object "
+            "'content_object'."
+        )
+        with self.assertRaisesMessage(ValueError, msg):
+            tagged_item.save()
+
+    def test_unsaved_instance_on_generic_foreign_key_bulk_create(self):
+        # Assigning an unsaved object to a GFK should raise a ValueError on bulk create.
         quartz = Mineral(name="Quartz", hardness=7)
-        with self.assertRaises(IntegrityError):
-            TaggedItem.objects.create(tag="shiny", content_object=quartz)
+        tagged_item = TaggedItem(tag="shiny", content_object=quartz)
+        msg = (
+            "bulk_create() prohibited to prevent data loss due to unsaved related "
+            "object 'content_object'."
+        )
+        with self.assertRaisesMessage(ValueError, msg):
+            TaggedItem.objects.bulk_create([tagged_item])
```

Comment:
We could add a similar test for `bulk_update()`.

Reply:
https://github.com/django/django/blob/6f311c7e35090957dd93783b71c5d3af7d5329d4/django/db/models/query.py#L736

We currently cannot bulk_update GenericForeignKeys as they are not a concrete fields (I could change this)

https://github.com/django/django/blob/6f311c7e35090957dd93783b71c5d3af7d5329d4/tests/queries/test_bulk_update.py#L166

I can explicitly add a GenericForeignKey case explicitly to this test if we want?

Reply:
Ahh true, sorry for the noise. No changes are required.

Reply:
Ah I added something because I had caused a problem not handling the fields being specified in bulk_update - let me know if the test is useful or too much

Reply:
I removed it.